programmers/완주하지-못한-선수.js

```diff
@@ -0,0 +1,15 @@
+function solution(participant, completion) {
+  const marathon = {};
+
+  for (let player of participant) {
+    player in marathon ? (marathon[player] += 1) : (marathon[player] = 1);
+  }
```

Comment:
변수명은 제 나름대로 고민해보고 좋아보이는 것으로 한번 바꿔봤어요
이게 좋다는 것은 아니고 제 생각이니 참고 정도만 해주세요 :) (marathon -> players)
```suggestion
  participant.forEach((part) => {
    return (part in players) ? (players[part] += 1) : (players[part] = 1);
  });
```
forEach나 map을 사용해서 반복문을 돌릴 수도 있어요

forEach와 map의 차이는 새로운 배열을 반환하냐 그렇지 않느냐에 있습니다.

**map**
```
const arr = [1, 2, 3];
const mapArr = arr.map((num) => num + 1); 
console.log(mapArr); // [2, 3, 4]
```

***forEach***
```
const arr = [1, 2, 3];
const forEachArr = [];
arr.forEach((num) => forEachArr.push(num + 1));
console.log(forEachArr) // [2, 3, 4];
```
***forEach를 map처럼 사용하면?***
```
const arr = [1, 2, 3];
const forEachArr = arr.forEach((num) => num + 1);
console.log(forEachArr) // undefined